### PR TITLE
Don't enable dl use in static builds

### PR DESCRIPTION
**Pull request by diyessi**
remove dladdr in static

### CODE REVIEW

File: src/ngraph/runtime/backend.cpp [hunk 1]
```diff
@@ -37,6 +37,9 @@ std::string runtime::Backend::s_backend_shared_library_search_directory;
 // This finds the full path of the containing shared library
 static string find_my_pathname()
 {
+#ifdef NGRAPH_STATIC_LIB_ENABLE
```
Comment (nmostafa) [inline src/ngraph/runtime/backend.cpp]: So we cannot dyn load any other backends besides CPU and INTERPRETER if the flag is enabled ?

File: src/ngraph/runtime/backend_manager.cpp [hunk 2]
```diff
@@ -32,12 +32,18 @@
 using namespace std;
 using namespace ngraph;
 
+#ifdef NGRAPH_STATIC_LIB_ENABLE
+#define DLERROR() ""
+#else
 #ifdef _WIN32
 #define CLOSE_LIBRARY(a) FreeLibrary(a)
 #define DLSYM(a, b) GetProcAddress(a, b)
+#define DLERROR() dlerror()
```
Comment (s-barannikov) [inline src/ngraph/runtime/backend_manager.cpp]: This should be in the other branch of the `#if`.

Comment (diyessi) [reply src/ngraph/runtime/backend_manager.cpp]: It's in all three branches; it's a NO-OP not on _WIN32.

Comment (s-barannikov) [reply src/ngraph/runtime/backend_manager.cpp]: That's confusing. Didn't know there is `dlerror` function on Windows.

Comment (s-barannikov) [reply src/ngraph/runtime/backend_manager.cpp]: Just to be clear: I'm talking about the implementation of the macro. `dlerror` sounds more posix-like.

File: src/ngraph/runtime/backend.cpp (hunk 1 above)
Comment (rkimballn1) [reply src/ngraph/runtime/backend.cpp]: Agreed. This should be fixed to work correctly.

File: src/ngraph/runtime/backend_manager.cpp (hunk 2 above)
Comment (rkimballn1) [inline src/ngraph/runtime/backend_manager.cpp]: dlerror() is linux not windows. Linux has the empty macro and windows has dlerror()

File: src/ngraph/runtime/backend.cpp (hunk 1 above)
Comment (diyessi) [reply src/ngraph/runtime/backend.cpp]: And what is "correctly"? There's no real documentation for what these are supposed to do.

**Comment by diyessi** [comment]
Duplicated by #3394

**Pull request closed without merging**